lib/utils/test-page-helper.js

```diff
@@ -123,18 +123,22 @@ function combineOptionValueIntoArray(optionValue) {
 }
 
 /**
- * Returns the browserId within the testPage string
+ * Returns the browserId of launcher
  *
- * @param {string} testPage
+ * @param {Object} launcher
  * @return {string}
  */
-function getBrowserId(testPage) {
-  const browserId = /browser=\s*([0-9]*)/.exec(testPage);
+function getBrowserId(launcher) {
+  try {
+    const testPage = launcher.settings.test_page;
+    const browserIdMatch =/browser=\s*([0-9]*)/.exec(testPage);
 
-  if (Array.isArray(browserId) !== null && browserId !== null) {
-    return browserId[1];
-  } else {
-    throw new Error('Browser Id can\'t be found.');
+    if (Array.isArray(browserIdMatch) !== null && browserIdMatch !== null) {
+      return browserIdMatch[1];
+    }
+  } catch(err) {
+    const errMsg = `${err.message} \n${err.stack} \nLauncher Settings: ${JSON.stringify(launcher.settings, null, 2)}`;
```

Comment:
I like that the error message provides the settings that was generated the error!!

Reply:
Reopened this, since I was unsure if this was resolved. @choheekim could you comment on this?

Reply:
the test’s checking whether it contains a given regex. As the error message can be vary I thought it’d be better to test against some message that will always be there.

Reply:
Got it. That makes sense. So if we disregard the test, were you able to test this path such that `err.message` provides a meaningful message?

Reply:
Yes, I was able to test it manually. 
e.g. the error message would look something like below:
```
Cannot read property 'test_page' of undefined
TypeError: Cannot read property 'test_page' of undefined
    at getBrowserId (/Users/cokim/WODS/ember-exam/lib/utils/test-page-helper.js:45:148)
   ...
Launcher Settings: undefined
```